### PR TITLE
[Cp] analytics update

### PR DESCRIPTION
**Pull request by eliasyishak**
Cherry pick to allow the new `package:unified_analytics` to handle the opt out if flutter users were already opted out in the legacy analytics system

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/runner/runner_test.dart [hunk 1]
```diff
@@ -316,28 +316,37 @@ void main() {
...
-
-      await runner.run(
-        <String>['--disable-telemetry'],
-        () => <FlutterCommand>[],
-        // This flutterVersion disables crash reporting.
-        flutterVersion: '[user-branch]/',
-        shutdownHooks: ShutdownHooks(),
-      );
+  group('unified_analytics', () {
+    final Usage legacyAnalytics = TestUsage();
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/runner/runner_test.dart:320]: Does this ever get used in the test?

Comment (eliasyishak) [reply packages/flutter_tools/test/general.shard/runner/runner_test.dart:320]: Yes, I'm using it in the setup to disable analytics

File: packages/flutter_tools/lib/runner.dart [hunk 2]
```diff
@@ -62,19 +63,6 @@ Future<int> run(
     StackTrace? firstStackTrace;
     return runZoned<Future<int>>(() async {
       try {
-        // Ensure that the consent message has been displayed
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/runner.dart:65]: Do we need to also cherrypick package:unified_analytics to get this functionality, or is it already in version 1.0.1?

Comment (eliasyishak) [reply packages/flutter_tools/lib/runner.dart:65]: Oh I didn't realize that the pubspec's would also need to be manually cherrypicked, but yes, we will need the latest `1.1.0` in the pubspec for this functionality. What's the approach for bumping the package version? Cherry pick the engine roll commit or can I just bump it in this cherrypick?

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**